pip_audit/_dependency_source/requirement.py

```diff
@@ -44,6 +53,7 @@ def __init__(
         index_url: str | None = None,
         extra_index_urls: list[str] = [],
         state: AuditState = AuditState(),
+        with_pip_compile: bool = False,
```

Comment:
Same comment as above: I don't think we want to imply that this is unique to `pip-compile`, since it wasn't before.